src/Illuminate/Support/ValidatedInput.php

```diff
@@ -183,137 +41,37 @@ public function merge(array $items)
-    }
-
     /**
      * Get the raw, underlying input array.
      *
+     * @param  array|mixed|null  $keys
      * @return array
      */
-    public function all()
+    public function all($keys = null)
     {
-        return $this->input;
-    }
-
-    /**
-     * Apply the callback if the validated inputs contains the given input item key.
-     *
-     * @param  string  $key
-     * @param  callable  $callback
-     * @param  callable|null  $default
-     * @return $this|mixed
-     */
-    public function whenHas($key, callable $callback, ?callable $default = null)
-    {
-        if ($this->has($key)) {
-            return $callback(data_get($this->all(), $key)) ?: $this;
-        }
-
-        if ($default) {
-            return $default();
-        }
-
-        return $this;
-    }
-
-    /**
-     * Determine if the validated inputs contains a non-empty value for an input item.
-     *
-     * @param  string|array  $key
-     * @return bool
-     */
-    public function filled($key)
-    {
-        $keys = is_array($key) ? $key : func_get_args();
-
-        foreach ($keys as $value) {
-            if ($this->isEmptyString($value)) {
-                return false;
-            }
-        }
-
-        return true;
-    }
-
-    /**
-     * Determine if the validated inputs contains an empty value for an input item.
-     *
-     * @param  string|array  $key
-     * @return bool
-     */
-    public function isNotFilled($key)
-    {
-        $keys = is_array($key) ? $key : func_get_args();
-
-        foreach ($keys as $value) {
-            if (! $this->isEmptyString($value)) {
-                return false;
-            }
+        if (! $keys) {
+            return $this->input;
         }
 
-        return true;
-    }
-
-    /**
-     * Determine if the validated inputs contains a non-empty value for any of the given inputs.
-     *
-     * @param  string|array  $keys
-     * @return bool
-     */
-    public function anyFilled($keys)
-    {
-        $keys = is_array($keys) ? $keys : func_get_args();
-
-        foreach ($keys as $key) {
-            if ($this->filled($key)) {
-                return true;
-            }
-        }
+        $input = [];
 
-        return false;
-    }
-
-    /**
-     * Apply the callback if the validated inputs contains a non-empty value for the given input item key.
-     *
-     * @param  string  $key
-     * @param  callable  $callback
-     * @param  callable|null  $default
-     * @return $this|mixed
-     */
-    public function whenFilled($key, callable $callback, ?callable $default = null)
-    {
-        if ($this->filled($key)) {
-            return $callback(data_get($this->all(), $key)) ?: $this;
-        }
-
-        if ($default) {
-            return $default();
+        foreach (is_array($keys) ? $keys : func_get_args() as $key) {
+            Arr::set($input, $key, Arr::get($this->input, $key));
         }
 
-        return $this;
+        return $input;
     }
```

Comment:
Is this change in behavior intentional? I mean, it offers to return "all" but sometimes returns just _some_. Wouldn't that work better with `only`?

Reply:
Yes, otherwise the abstract `all` definition on the trait and class are incompatible and PHP will throw an exception.

Reply:
This is also basically the same definition and logic that exists on the `InteractsWithInput` trait:

https://github.com/laravel/framework/blob/764c79608eb29351d9f0d2c8208eba36d0422db6/src/Illuminate/Http/Concerns/InteractsWithInput.php#L80-L101

And `UriQueryString`:

https://github.com/laravel/framework/blob/cf7886bdee7017081263b14510c8d4753aa17a52/src/Illuminate/Support/UriQueryString.php#L22-L43

Reply:
Ah, okay, I see 😬 Thanks for the clarification 👍🏻

Reply:
Maybe we should modify this in Laravel 12 and remove the $keys parameter in favour of a standard `only` method?

Not sure... Might not be worth the breaking changes...

Reply:
It would be _right_ but probably _denied_